js/components/graph/FocusBar.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from "prop-types";
+import FocusTab from "./FocusTab.js";
+
+// These lists are in reverse order to what ends up appearing on the screen
+const computerScienceFocusLabels = [
+  ["web", "Web Technologies"],
+  ["theory", "Theory of Computation"],
+  ["HCI", "HumanComp Interaction"],
+  ["game", "Video Games"],
+  ["systems", "Computer Systems"],
+  ["vision", "Computer Vision"],
+  ["NLP", "Computational Linguistics"],
+  ["AI", "Artificial Intelligence"],
+  ["sci", "Scientific Computing"],
+];
+
+/**
+ * React component representing the focus menu bar
+ */
+export default class FocusBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+    };
+  }
+
+  /**
+   * Changes whether the focus bar is open or not
+   */
+  toggleFocusBar = () => {
+    if (this.state.open) {
+      this.setState({open: false});
+    } else {
+      this.setState({open: true});
+    }
+}
+
+  /**
+   * Creates the menu items of the focus bar using the FocusTab component
+   * @returns an array of FocusTab components
+   */
+  generateFocusTabs = () => {
+    return computerScienceFocusLabels.map((focus) => {
```

Comment:
Hmm, here I believe you can use list destructuring for the parameter, something like `[focusId, focusTitle] => { ... }`. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment.